cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/handlers/AuthenticationHandler.java

```diff
@@ -99,6 +100,11 @@ public AuthenticationHandler(CConfiguration cConf, SConfiguration sConf,
     this.userIdentityExtractor = userIdentityExtractor;
   }
 
+  private boolean propagateUserCredential(CConfiguration cConf) {
+    return cConf.getBoolean(Constants.Security.Authentication.PROPAGATE_USER_CREDENTIAL)
+      || SecurityUtil.isInternalAuthEnabled(cConf);
```

Comment:
Let's avoid propagating credential based on either or. Only propagate when config propagate is on

Reply:
In this situation, internal auth doesn't work at all if credential propagation (or credential-less auth) is used. Went ahead and switched this check to be performed on the access enforcer extension instead.